src/server/jobs/process-creator-program-early-access-rewards.ts

```diff
@@ -0,0 +1,147 @@
+      JOIN "Model" m ON mv."modelId" = m.id
+      WHERE mv."status" = 'Published'
+        AND mv."earlyAccessTimeFrame" > 0
+        AND m."userId" IN (${Prisma.join(creatorProgramUserIds, ',')})
+        AND GREATEST(mv."createdAt", mv."publishedAt") 
+          + (mv."earlyAccessTimeFrame" || ' day')::INTERVAL
+          >= ${lastUpdate};
+    `;
+
+    if (modelVersions.length === 0) {
+      await setLastUpdate();
+      return; // No records to process
+    }
+
+    // Get all records that need to be processed
+    const modelVersionData = await clickhouse
+      .query({
+        query: `
+          SELECT
+            modelVersionId,
+            createdDate,
+            uniqMerge(users_state) AS downloads
+          FROM daily_downloads_unique_mv
+          WHERE modelVersionId IN (${modelVersions.map(({ id }) => id).join(',')})
+          AND createdDate > subtractDays(toStartOfDay(now()), 14)
+          GROUP BY modelVersionId, createdDate
+          ORDER BY createdDate DESC;
+    `,
+        format: 'JSONEachRow',
+      })
+      .then((x) => x.json<{ modelVersionId: number; createdDate: Date; downloads: number }[]>());
+
+    await Promise.all(
+      modelVersions.map(async (version) => {
+        // First, check that it's still early access:
+        const isEarlyAccessBool = isEarlyAccess({
+          versionCreatedAt: version.createdAt,
+          earlyAccessTimeframe: version.earlyAccessTimeFrame,
+          publishedAt: version.publishedAt,
+        });
+
+        const downloadData = modelVersionData
+          .filter(
+            (x) =>
+              x.modelVersionId === version.id &&
+              dayjs(x.createdDate).endOf('day').isAfter(version.publishedAt) &&
+              dayjs(x.createdDate)
+                .startOf('day')
+                .isBefore(dayjs(version.publishedAt).add(version.earlyAccessTimeFrame, 'day'))
+          )
+          .map((d) => ({
+            date: d.createdDate,
+            downloads: Number(d.downloads ?? '0'),
+          }));
+
+        if (downloadData.length === 0) {
+          return;
+        }
+
+        if (!isEarlyAccessBool) {
+          // apply the reward:
+          const totalDownloads = downloadData.reduce((acc, x) => acc + Number(x.downloads), 0);
+
+          await createBuzzTransaction({
+            fromAccountId: 0,
+            toAccountId: version.userId,
+            amount: totalDownloads * 10, // 10 buzz per download
+            description: `Early access reward - ${version.modelName} - ${version.modelVersionName}`,
+            type: TransactionType.Reward,
+            externalTransactionId: `model-version-${version.id}-early-access-reward`,
+          });
+        }
+
+        await dbWrite.modelVersion.update({
+          where: { id: version.id },
+          data: {
+            meta: {
+              ...version.meta,
+              earlyAccessDownloadData: downloadData,
+            },
+          },
+        });
+      })
+    );
```

Comment:
We might switch this to do an appending update rather than a full replace. Here's how we do it in the user service for settings for example:
```
UPDATE "User" SET settings = COALESCE(settings,'{}') || '${JSON.stringify(toSet)}'::jsonb WHERE id = ${userId}
```